lib/messages.js

```diff
@@ -22,33 +22,39 @@ var Messages = {
     var storedMessages = this.getStored();
     this.clearList();
     for (var i = 0; i < storedMessages.length; i++) {
-      this.render(storedMessages[i].text);
+      this.render(storedMessages[i]);
     }
   },
 
   clearList: function() {
     $messageList.html('');
   },
 
-  render: function(messageOutput) {
-    $messageList.append(`
-      <article class="message-js">
-        <p class="user-message" contenteditable="true">` + messageOutput + `</p>
-        <button class="delete-button-js"></button>
-      </article>
+  render: function(messageObject) {
+    if (messageObject.user === 'user') {
+      $messageList.append(`
+        <article class="message-js" id="${messageObject.id}">
+          <p class="user-message" contenteditable="true">${messageObject.text}</p>
+          <button class="delete-button-js"></button>
+        </article>
+        `);
+    } else {
+      $messageList.append(`
+        <article class="robot-message-js" id="${messageObject.id}">
+          <p>${messageObject.text}</p>
+        </article>
       `);
+    }
   },
 
-  renderRobot: function(robotOutput) {
-    $messageList.append(`
-      <article class="robot-message-js">
-        <p>` + robotOutput + `</p>
-      </article>
-    `);
-  },
-
-  deleteFromStorage: function() {
-
+  deleteFromStorage: function(idToBeDeleted) {
+    var id = parseInt(idToBeDeleted);
+    for (var i = 0; i < this.allMessages.length; i++) {
+      if(this.allMessages[i].id === id) {
+        this.allMessages.splice(i, 1);
+      }
+    }
+    this.addToArray();
```

Comment:
I think we can use .filter here to update the array without having to use a for loop.